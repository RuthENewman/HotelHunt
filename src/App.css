*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

html {
  font-size: 62.5%; /* 1rem = 10px, 10px / 16px = 62.5%; */

}

body {
  box-sizing: border-box;
  font-family: 'Lato', sans-serif;
  padding: 3rem;
  background-size: cover;
}

/* Header section */

.header {
  color: white;
  text-align: center;
  background-image: linear-gradient(to right bottom,
    rgba(164,251,237, 0.7),
    rgba(248,7,89, 0.7)),
  url('./images/hero.jpg');
  height: 35rem;
  background-size: cover;
  padding: 14rem;
}

.heading-main--large {
  font-size: 5.2rem;
  letter-spacing: 1rem;
  text-transform: uppercase;
  margin-bottom: 2rem;
}

.heading-main--sub {
  font-size: 3.1rem;
  letter-spacing: .5rem;
}

/* Form section */

.facilities-form {
  position: relative;
  margin: 0 auto;
  width: 100%;
  font-size: 1.8rem;
  background-image: linear-gradient(to right bottom,
    rgba(228,18,96, 1),
    rgba(248,7,89, 1));
  padding: 2rem 0;
  text-align: center;
  margin-bottom: 1rem;
}

.facilities-form__buttons {

}

[class^="facilities-form__buttons-"] {
  display: inline-block;
  padding: .8rem;
  background-color: white;
  border-radius: .3rem;
  margin: 0 .5rem;
}

label {
  margin-right: .2rem;
  padding: 1rem;
  color: #808080;
}

.input-button {
  /* display: none; */
}

#submitButton {
  text-transform: uppercase;
  text-decoration: none;
  padding: 1rem 3rem;
  border-radius: 0.5rem;
  display: inline-block;
  position: relative;
  font-size: 2rem;
  border: none;
  color: white;
  background-color: #bc4e9c;
  margin-top: 1rem;
}

#submitButton:hover {
  cursor: pointer;
  width: 15rem;
  opacity: .8;
}

/* Results section */

.results {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  max-width: 100%;
}

.hotel {
  margin: 0 2rem;
  width: 30%;
}

.hotel__details {
  margin: 1rem 0;
}

.hotel__details--name {
  font-size: 3rem;
  display: inline-block;
  margin-right: 1rem;
}

.hotel__details--starRating {
  font-size: 1.8rem;
  display: inline-block;
  color: #ff8c00;
}

.hotel__image {
  opacity: .5;
  border-radius: .4rem;
  width: 100%;
}

.hotel__image:hover {
  opacity: 1;

}
