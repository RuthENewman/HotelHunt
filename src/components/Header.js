import React, {Component} from 'react';

class Header extends Component {

  render() {
    return (
      <header className="header">
        <h1 className="heading-main--large">
          Hotel hunt
        </h1>
        <h2 className="heading-main--sub">
          Find by facilities
        </h2>
    </header>
    )
  }

}

export default Header;
