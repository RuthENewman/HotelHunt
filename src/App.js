import React, {Component} from 'react';
import Header from './components/Header';
import Form from './components/Form';
import Results from './components/Results';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props)

    this.state = {
      hotels: [],
      filteredHotels: [],
      filters: {
        wifi: false,
        pool: false,
        restaurant: false,
        carPark: false,
        gym: false,
        spa: false,
      }
    }
  }

  render() {
    return (
    <div className="App">
      <Header />
      <Form />
      <Results />
    </div>
  );
  }
}

export default App;
