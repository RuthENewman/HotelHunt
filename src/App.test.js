import React from 'react';
import ReactDOM from 'react-dom';
import App,  { hotels } from './App';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from './components/Header';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should render Header component correctly', () => {
   const renderer = new ReactShallowRenderer();
   renderer.render(<Header />);
   expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should fetch all hotels', () => {
    const result = hotels;
   expect(result.hotels.length).toBe(10);
});
