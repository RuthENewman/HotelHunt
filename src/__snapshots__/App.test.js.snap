// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<header
  className="header"
>
  <h1
    className="heading-main--large"
  >
    Hotel hunt
  </h1>
  <h2
    className="heading-main--sub"
  >
    Find by facilities
  </h2>
</header>
`;
